### PR TITLE
Remove log on different source

### PR DESCRIPTION
**Pull request by siscia**
This is the very last PR, built on top of #2467 

Here we use PANIC also for logs that are not directed to `kLogCvmfs`.

This completely address [CVM-1697](https://sft.its.cern.ch/jira/projects/CVM/issues/CVM-1697)

There is this line here: https://github.com/cvmfs/cvmfs/blob/devel/cvmfs/options.cc#L194 where I would like a second opinion.

I believe it is fine as it is, without need to be modified.

### CODE REVIEW

File: cvmfs/signing_tool.cc [hunk 1]
```diff
@@ -230,8 +231,7 @@ SigningTool::Result SigningTool::Run(
       reinterpret_cast<const unsigned char *>(published_hash.ToString().data()),
       published_hash.GetHexSize(), &sig, &sig_size);
   if (!manifest_was_signed) {
-    abort();
-    LogCvmfs(kLogCvmfs, kLogStderr, "Failed to sign manifest");
+    PANIC(kLogStderr, "Failed to sign manifest");
     return kError;
```
Comment (jblomer) [inline cvmfs/signing_tool.cc]: This can be removed along the way

File: cvmfs/sync_item.cc [hunk 2]
```diff
@@ -67,10 +68,11 @@ SyncItemType SyncItem::GetGenericFiletype(const SyncItem::EntryStat &stat) const
 {
   const SyncItemType type = stat.GetSyncItemType();
   if (type == kItemUnknown) {
-    PrintWarning("'" + GetRelativePath() + "' has an unsupported file type "
-                 "(st_mode: " + StringifyInt(stat.stat.st_mode) +
-                 " errno: " + StringifyInt(stat.error_code) + ")");
-    abort();
+    PANIC(kLogStderr, ("[WARNING] '" + GetRelativePath() +
```
Comment (jblomer) [inline cvmfs/sync_item.cc]: Should be transformed into a proper format string

File: cvmfs/sync_item.cc [hunk 3]
```diff
@@ -94,9 +96,10 @@ SyncItemType SyncItem::GetRdOnlyFiletype() const {
 SyncItemType SyncItemNative::GetScratchFiletype() const {
   StatScratch();
   if (scratch_stat_.error_code != 0) {
-    PrintWarning("Failed to stat() '" + GetRelativePath() + "' in scratch. "
-                 "(errno: " + StringifyInt(scratch_stat_.error_code) + ")");
-    abort();
+    PANIC(kLogStderr, ("[WARNING] Failed to stat() '" + GetRelativePath() +
```
Comment (jblomer) [inline cvmfs/sync_item.cc]: Make format string

File: cvmfs/sync_mediator.cc [hunk 4]
```diff
@@ -115,8 +116,8 @@ void SyncMediator::EnsureAllowed(SharedPtr<SyncItem> entry) {
                   string(catalog::VirtualCatalog::kVirtualPath) + "/",
                   ignore_case_setting)) )
   {
-    PrintError("invalid attempt to modify '" + relative_path + "'");
-    abort();
+    PANIC(kLogStderr, "[ERROR] invalid attempt to modify '",
```
Comment (jblomer) [inline cvmfs/sync_mediator.cc]: `%s` missing

File: cvmfs/upload_gateway.cc [hunk 5]
```diff
@@ -60,7 +61,7 @@ GatewayUploader::GatewayUploader(const SpoolerDefinition& spooler_definition)
          spooler_definition.driver_type == SpoolerDefinition::Gateway);
 
   if (!ParseSpoolerDefinition(spooler_definition, &config_)) {
-    abort();
+    PANIC(NULL);
```
Comment (jblomer) [inline cvmfs/upload_gateway.cc]: Let's add a message here

File: cvmfs/upload_s3.cc [hunk 6]
```diff
@@ -49,7 +50,7 @@ S3Uploader::S3Uploader(const SpoolerDefinition &spooler_definition)
   atomic_init32(&terminate_);
 
   if (!ParseSpoolerDefinition(spooler_definition)) {
-    abort();
+    PANIC(NULL);
```
Comment (jblomer) [inline cvmfs/upload_s3.cc]: Let's add a message

File: cvmfs/util/exception.cc [hunk 7]
```diff
@@ -47,6 +47,16 @@ void Panic(const char* coordinates, const LogSource source, const int mask,
 #endif
 }
 
+void Panic(const char* coordinates, const LogSource _source, const int _mask) {
+  assert(_mask == 0);
```
Comment (jblomer) [inline cvmfs/util/exception.cc]: Shouldn't we simply call `Panic(coordinates, _source, _mask, "")`?

File: cvmfs/util/exception.h [hunk 8]
```diff
@@ -24,6 +24,7 @@ class ECvmfsException : std::runtime_error {
 
 void Panic(const char *coordinates, const LogSource source, const int mask,
            const char *format, ...);
+void Panic(const char *coordinates, const LogSource source, const int mast);
```
Comment (jblomer) [inline cvmfs/util/exception.h]: Please add a unit test for the `PANIC(NULL)` invocation

File: cvmfs/authz/authz_fetch.cc [hunk 9]
```diff
@@ -170,9 +171,8 @@ void AuthzExternalFetcher::ExecHelper() {
       close(fd);
 
     execve(argv0, argv, &envp[0]);
-    syslog(LOG_USER | LOG_ERR, "failed to start authz helper %s (%d)",
-           argv0, errno);
-    abort();
+    PANIC(kLogStdout | kLogStderr, "failed to start authz helper %s (%d)",
```
[the remark this answers is not in the document]
Comment (siscia) [reply cvmfs/authz/authz_fetch.cc]: Reverted!

File: cvmfs/CMakeLists.txt [hunk 10]
```diff
@@ -447,7 +455,10 @@ set (CVMFS_SWISSKNIFE_SOURCES
 set (CVMFS_SUID_HELPER_SOURCES
   cvmfs_suid_helper.cc
   cvmfs_suid_util.cc
+  logging.cc
```
[the remark this answers is not in the document]
Comment (siscia) [reply cvmfs/CMakeLists.txt]: It was an unused indirect dependency.

**Comment by jblomer** [comment]
Please have a look at the linter error

**Comment by siscia** [comment]
Sorry, I pushed the fix on the wrong remote!

It should be fine now!

File: cvmfs/CMakeLists.txt [hunk 11]
```diff
@@ -924,6 +932,7 @@ if (BUILD_SERVER)
                          ${OPENSSL_LIBRARIES} ${RT_LIBRARY} ${VJSON_LIBRARIES}
                          ${SHA3_LIBRARIES} ${CAP_LIBRARIES} ${LibArchive_LIBRARY}
                          pthread dl)
+  target_link_libraries (cvmfs_suid_helper pthread)
```
Comment (jblomer) [inline cvmfs/CMakeLists.txt]: This isn't need either, is it?

File: cvmfs/sanitizer.cc [hunk 12]
```diff
@@ -4,11 +4,13 @@
  * Provides input data sanitizer in the form of whitelist of character ranges.
  */
 
-#include "cvmfs_config.h"
 #include "sanitizer.h"
 
 #include <cassert>
 
+#include "cvmfs_config.h"
+#include "util/exception.h"
```
Comment (jblomer) [inline cvmfs/sanitizer.cc]: This include should be removed

File: cvmfs/sanitizer.cc [hunk 13]
```diff
@@ -4,11 +4,13 @@
  * Provides input data sanitizer in the form of whitelist of character ranges.
  */
 
-#include "cvmfs_config.h"
 #include "sanitizer.h"
 
 #include <cassert>
 
+#include "cvmfs_config.h"
```
Comment (jblomer) [inline cvmfs/sanitizer.cc]: This should be the first include

File: cvmfs/authz/authz_fetch.cc [hunk 14]
```diff
@@ -21,6 +21,7 @@
 #include "platform.h"
 #include "sanitizer.h"
 #include "smalloc.h"
+#include "util/exception.h"
```
Comment (jblomer) [inline cvmfs/authz/authz_fetch.cc]: I think we can remove the include, too
